File: easy/problems/0374-guess-number-higher-or-lower/solution.cpp
class Solution : public GuessGame {
public:
    int guessNumber(int n) {
        int left = 1, right = n;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            int result = guess(mid);

            if (result == 0) {
                return mid;
            } else if (result == -1) {
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return -1;
    }
};

File: easy/problems/0374-guess-number-higher-or-lower/solution.py
def guessNumber(n):
    """
    Binary search to find the guessed number.
    Uses the guess(x) API function provided by LeetCode.

    Args:
        n: int - upper bound of range

    Returns:
        int - the guessed number
    """
    left, right = 1, n

    while left <= right:
        mid = (left + right) // 2
        result = guess(mid)

        if result == 0:
            return mid
        elif result == -1:
            right = mid - 1
        else:
            left = mid + 1
